packages/cli/src/execution-lifecycle/save-execution-progress.ts

```diff
@@ -78,27 +63,19 @@ export async function saveExecutionProgress(
 
 		fullExecutionData.status = 'running';
 
-		await Container.get(ExecutionRepository).updateExistingExecution(
-			executionId,
-			fullExecutionData,
-		);
+		await executionRepository.updateExistingExecution(executionId, fullExecutionData);
 	} catch (e) {
 		const error = e instanceof Error ? e : new Error(`${e}`);
 
-		Container.get(ErrorReporter).error(error);
+		errorReporter.error(error);
```

Comment:
Do we still need then TODO below?

Reply:
not sure. what do you think?

Reply:
I'm not clear on what the TODO is suggesting here. In other hooks, when a DB write throws, we report it and log it and move on.

Reply:
TBH, neither am I. I just did not want to change anything that I did not understand just yet.